src/test/scala/edu/ie3/simona/model/participant/SystemParticipantSpec.scala

```diff
@@ -0,0 +1,266 @@
+      qCalc should approximate(qSol)
+
+    }
+  }
+
+  "calculate reactive power correctly for a standard q_v characteristic" in {
+    val loadMock = new MockParticipant(
+      UUID.fromString("d8461624-d142-4360-8e02-c21965ec555e"),
+      "System participant calculateQ Test",
+      OperationInterval(0L, 86400L),
+      QControl(new QV("qV:{(0.93,-1),(0.97,0),(1,0),(1.03,0),(1.07,1)}")),
+      Kilowatts(200),
+      0.98,
+    )
+
+    val testCases = Table(
+      ("adjustedVoltageVal", "qSol"),
+      (0.9, Kilovars(-39.799)),
+      (0.93, Kilovars(-39.799)),
+      (0.95, Kilovars(-19.899)),
+      (0.97, Kilovars(0)),
+      (1.00, Kilovars(0)),
+      (1.03, Kilovars(0)),
+      (1.05, Kilovars(19.899)),
+      (1.07, Kilovars(39.799)),
+      (1.1, Kilovars(39.799)),
+    )
+
+    forAll(testCases) { (adjustedVoltageVal, qSol) =>
+      val adjustedVoltage = Each(adjustedVoltageVal)
+      val p = Kilowatts(42)
+      val qCalc = loadMock.calculateReactivePower(p, adjustedVoltage)
+      qCalc should approximate(qSol)
+    }
+  }
+
+  "calculate reactive power correctly for q_v characteristic if active power is zero and cosPhiRated is 1" in {
+    val loadMock = new MockParticipant(
+      UUID.fromString("d8461624-d142-4360-8e02-c21965ec555e"),
+      "System participant calculateQ Test",
+      OperationInterval(0L, 86400L),
+      QControl(new QV("qV:{(0.93,-1),(0.97,0),(1,0),(1.03,0),(1.07,1)}")),
+      Kilowatts(200),
+      1d,
+    )
+
+    val testCases = Table(
+      ("adjustedVoltageVal", "qSol"),
+      (0.9, Kilovars(0)),
+      (0.93, Kilovars(0)),
+      (0.95, Kilovars(0)),
+      (0.97, Kilovars(0)),
+      (1.00, Kilovars(0)),
+      (1.03, Kilovars(0)),
+      (1.05, Kilovars(0)),
+      (1.07, Kilovars(0)),
+      (1.1, Kilovars(0)),
+    )
+
+    forAll(testCases) { (adjustedVoltageVal, qSol) =>
+      val adjustedVoltage = Each(adjustedVoltageVal)
+      val p = Kilowatts(0)
+      val qCalc = loadMock.calculateReactivePower(p, adjustedVoltage)
+      qCalc should approximate(qSol)
+    }
+  }
+
+  "calculate reactive power correctly for q_v characteristic if active power is not zero and cosPhiRated is 0.95" in {
+    val loadMock = new MockParticipant(
+      UUID.fromString("d8461624-d142-4360-8e02-c21965ec555e"),
+      "System participant calculateQ Test",
+      OperationInterval(0L, 86400L),
+      QControl(new QV("qV:{(0.93,-1),(0.97,0),(1,0),(1.03,0),(1.07,1)}")),
+      Kilowatts(200),
+      0.95,
+    )
+
+    val testCases = Table(
+      ("adjustedVoltageVal", "qSol"),
+      (0.9, Kilovars(-62.44)),
+      (0.93, Kilovars(-62.44)),
+      (0.95, Kilovars(-31.22)),
+      (0.97, Kilovars(0)),
+      (1.00, Kilovars(0)),
+      (1.03, Kilovars(0)),
+      (1.05, Kilovars(31.22)),
+      (1.07, Kilovars(62.44)),
+      (1.1, Kilovars(62.44)),
+    )
+
+    forAll(testCases) { (adjustedVoltageVal, expectedQ) =>
+      val adjustedVoltage = Each(adjustedVoltageVal)
+      val p = Kilowatts(42)
```

Comment:
Please note, this was originally 100 kW. It is working correct as well at operation point of 42 kW but we should not change the test with this PR.